action/protocol/staking/protocol.go

```diff
@@ -114,3 +119,36 @@ func (p *Protocol) handleRestake(ctx context.Context, act action.Action, sm prot
 	// TODO
 	return nil, nil
 }
+
+func (p *Protocol) hasDelegate(name address.Address) bool {
```

Comment:
if this is a private function, why not call p.delegates.Contains(name) directly?